pyccel/codegen/cwrapper.py

```diff
@@ -22,23 +22,33 @@ def write_python_wrapper(expr, printer):
     results_decs = '\n    '.join(printer._print(i) for i in results_decs)
     code += '{0}\n    {1}\n    '.format(arg_decs, results_decs)
 
-    code += "if (!PyArg_ParseTuple(args, \""
-    code += ''.join(pytype_registry[str_dtype(arg.dtype)] for arg in expr.arguments)
-    code += "\", "
-    code += ', '.join("&" + printer._print(arg) for arg in expr.arguments)
-    code += "))\n        return NULL;\n    "
+    # Check if the function with No arguments
+    if not expr.arguments:
+        code += "if (!PyArg_ParseTuple(args, \"\"))\n        return NULL;\n    "
+    else:
+        code += "if (!PyArg_ParseTuple(args, \""
+        code += ''.join(pytype_registry[str_dtype(arg.dtype)] for arg in expr.arguments)
+        code += "\", "
+        code += ', '.join("&" + printer._print(arg) for arg in expr.arguments)
+        code += "))\n        return NULL;\n    "
 
     if len(expr.results)==0:
         func_call = UndefinedFunction(str(expr.name))(*expr.arguments)
```

Comment:
As described in issue #397 we should not be using `UndefinedFunction` here as it is a sympy object which does not contain information which may be necessary for printing. Please change this line as described in the issue